src/shogun/base/SGObject.cpp

```diff
@@ -815,33 +816,44 @@ void CSGObject::subscribe_to_parameters(ParameterObserverInterface* obs)
 
 	// Create an observable which emits values only if they are about
 	// parameters selected by the observable.
-	auto subscription =
-	    m_observable_params
-	        ->filter([obs](ObservedValue v) { return obs->filter(v.name); })
-	        .timestamp()
-	        .subscribe(sub);
+	auto subscription = m_observable_params
+	                        ->filter([obs](ObservedValue v) {
+		                        return obs->filter(v.get_name());
+		                    })
+	                        .timestamp()
+	                        .subscribe(sub);
 }
 
-void CSGObject::observe_scalar(
-    const int64_t step, const std::string& name, const Any& value)
+void CSGObject::observe(const ObservedValue* value)
 {
-	ObservedValue tmp;
-	tmp.step = step;
-	tmp.name = name;
-	tmp.value = value;
-	m_subscriber_params->on_next(tmp);
+	m_subscriber_params->on_next(*value);
 }
 
 class CSGObject::ParameterObserverList
 {
 public:
 	void register_param(
-	    const std::string& name, const std::string& type,
+	    const std::string& name, const SG_OBS_VALUE_TYPE type,
 	    const std::string& description)
 	{
 		m_list_obs_params[name] = std::make_pair(type, description);
 	}
 
+	std::string type_name(SG_OBS_VALUE_TYPE type)
```

Comment:
this is for debug purposes only?
or is there somewhere we use `type_name` internally?